core/src/main/java/org/apache/carbondata/core/scan/filter/executer/MeasureColumnExecuterFilterInfo.java

```diff
@@ -16,15 +16,29 @@
  */
 package org.apache.carbondata.core.scan.filter.executer;
 
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
 public class MeasureColumnExecuterFilterInfo {
 
   Object[] filterKeys;
+  Set filterKeysSet;
 
   public void setFilterKeys(Object[] filterKeys) {
     this.filterKeys = filterKeys;
   }
 
+  public void setFilterKeysSet(Object[] filterKeys) {
+    this.filterKeysSet = new HashSet<Object>(Arrays.asList(filterKeys));
```

Comment:
better to judge "filterKeys is null"

Reply:
it must contain null, because when the case is  sql with where field is null, it will also run this code.
if we remove null value, it can not filter null value, and the result will be wrong.

Reply:
I mean that if filterKeys is null, we should use ```Arrays.asList(filterKeys)```, we can use ``` new HashSet<Object>()``` directly, because ```Arrays.asList``` will throw NPE

Reply:
`      Object[] keysBasedOnFilter = filterValues.getMeasuresFilterValuesList()
          .toArray((new Object[filterValues.getMeasuresFilterValuesList().size()]));`
`msrColumnExecuterInfo.setFilteKeysSet(keysBasedOnFilter);`
the filterkeys  is nerver null. but the element in filterkeys can be null.